pkg/validating-webhooks/hive/v1/clusterdeploymentcustomization_validating_admission_hook.go

```diff
@@ -0,0 +1,285 @@
+	return &admissionv1beta1.AdmissionResponse{
+		Allowed: true,
+	}
+}
+
+// validateUpdate specifically validates update operations for ClusterDeployment objects.
+func (a *ClusterDeploymentCustomizationValidatingAdmissionHook) validateUpdate(admissionSpec *admissionv1beta1.AdmissionRequest) *admissionv1beta1.AdmissionResponse {
+	contextLogger := log.WithFields(log.Fields{
+		"operation": admissionSpec.Operation,
+		"group":     admissionSpec.Resource.Group,
+		"version":   admissionSpec.Resource.Version,
+		"resource":  admissionSpec.Resource.Resource,
+		"method":    "validateUpdate",
+	})
+
+	newObject := &hivev1.ClusterDeploymentCustomization{}
+	if err := a.decoder.DecodeRaw(admissionSpec.Object, newObject); err != nil {
+		contextLogger.Errorf("Failed unmarshaling Object: %v", err.Error())
+		return &admissionv1beta1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	// Add the new data to the contextLogger
+	contextLogger.Data["object.Name"] = newObject.Name
+
+	oldObject := &hivev1.ClusterDeploymentCustomization{}
+	if err := a.decoder.DecodeRaw(admissionSpec.OldObject, oldObject); err != nil {
+		contextLogger.Errorf("Failed unmarshaling OldObject: %v", err.Error())
+		return &admissionv1beta1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	// Add the new data to the contextLogger
+	contextLogger.Data["oldObject.Name"] = oldObject.Name
+
+	allErrs := field.ErrorList{}
+	specPath := field.NewPath("spec")
+
+	allErrs = append(allErrs, validateInstallConfigPatches(specPath, newObject.Spec.InstallConfigPatches)...)
+
+	if len(allErrs) > 0 {
+		contextLogger.WithError(allErrs.ToAggregate()).Info("failed validation")
+		status := errors.NewInvalid(schemaGVK(admissionSpec.Kind).GroupKind(), admissionSpec.Name, allErrs).Status()
+		return &admissionv1beta1.AdmissionResponse{
+			Allowed: false,
+			Result:  &status,
+		}
+	}
+
+	// If we get here, then all checks passed, so the object is valid.
+	contextLogger.Info("Successful validation")
+	return &admissionv1beta1.AdmissionResponse{
+		Allowed: true,
+	}
+}
+
+func validateInstallConfigPatches(path *field.Path, patches []hivev1.PatchEntity) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for i, patch := range patches {
+		if !isValidOP(patch.Op) {
+			allErrs = append(allErrs, field.Invalid(path.Index(i), patch, "install config patch op must be a valid json patch operation"))
+		}
+		if len(patch.Path) == 0 || !strings.HasPrefix(patch.Path, "/") {
+			allErrs = append(allErrs, field.Invalid(path.Index(i), patch, "install config patch path must start with '/'"))
+		}
+	}
+	return allErrs
+}
+
+func isValidOP(op string) bool {
```

Comment:
It would be neat if this used the `Op` [type](https://github.com/krishicks/yaml-patch/blob/7cea92d7f43e6d129121acd27b1d0063692e4039/operation.go#L9-L10) and [value consts](https://github.com/krishicks/yaml-patch/blob/7cea92d7f43e6d129121acd27b1d0063692e4039/operation.go#L12-L20) from yaml-patch. (It would be neater if yaml-patch provided a func to validate that a string was a valid `Op`... but that's a fight for another day.)